src/server.c

```diff
@@ -77,13 +80,13 @@ int dqlite__init(struct dqlite_node *d,
 #else
 	rv = sem_init(&d->ready, 0, 0);
 	if (rv != 0) {
-		/* TODO: better error reporting */
+		snprintf(d->errmsg, DQLITE_ERRMSG_BUF_SIZE, "sem_init(): failed");
```

Comment:
Here as well, something like:

```C
snprintf(d->errmsg, DQLITE_ERRMSG_BUF_SIZE, "sem_init(): %s", strerror(rv));
```

Reply:
strerror is not necessarily thread-safe, which seems like a concern here, so maybe strerror_r would be better? I see strerror is already used elsewhere.

Reply:
> strerror is not necessarily thread-safe, which seems like a concern here, so maybe strerror_r would be better? I see strerror is already used elsewhere.

In theory we could hit some races, if for example you run this code concurrently with other parts of your application, but that seems unlikely. That being said, using `strerror_r` would be certainly better.

Reply:
I'll use strerror for now and look into switching everything to strerror_r in a separate PR.